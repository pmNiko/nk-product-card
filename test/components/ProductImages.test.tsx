import React from 'react';
import rendered from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductImage } from '../../src/components/Productimage';
import { product1, product2 } from '../data/product';

describe('Test Product Images', () => {
  test('should display component with custom image', () => {
    const wrapper = rendered.create(
      <ProductCard product={product1}>
        {() => <ProductImage img="https://image.png" />}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should display component with the image product', () => {
    const wrapper = rendered.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
