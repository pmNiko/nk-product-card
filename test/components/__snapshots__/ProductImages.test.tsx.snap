// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Product Images should display component with custom image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="https://image.png"
  />
</div>
`;

exports[`Test Product Images should display component with the image product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
